src/power_grid_model/validation/errors.py

```diff
@@ -325,12 +325,14 @@ def __init__(  # pylint: disable=too-many-arguments
         self,
         component: ComponentType,
         field: str,
-        ids: list[int],
-        ref_components: ComponentType | list[ComponentType],
+        ids: Optional[list[int]] = None,
+        ref_components: Optional[ComponentType | list[ComponentType]] = None,
         filters: Optional[dict[str, Any]] = None,
     ):
         # pylint: disable=too-many-positional-arguments
-        super().__init__(component=component, field=field, ids=ids)
+        self.ids = ids if ids is not None else []
+        ref_components = ref_components if ref_components is not None else []
+        super().__init__(component=component, field=field, ids=self.ids)
```

Comment:
initialization order in python: either all before, or all after `super().__init__` but no mixing please